### PR TITLE
Implement `destination.json` schema v3.0

### PR DESCRIPTION
**Pull request by MaxDesiatov**
This brings the implementation in line with changes proposed in https://github.com/apple/swift-evolution/pull/1942.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Review by tomerd — APPROVED** [review]

File: Sources/CoreCommands/SwiftTool.swift [hunk 1]
```diff
@@ -654,17 +711,32 @@ public final class SwiftTool {
...
+            {
                 destination = targetDestination
             } else if let destinationSelector = options.build.crossCompilationDestinationSelector {
                 destination = try DestinationsBundle.selectDestination(
                     fromBundlesAt: sharedCrossCompilationDestinationsDirectory,
                     fileSystem: fileSystem,
                     matching: destinationSelector,
                     hostTriple: hostTriple,
-                    observabilityScope: observabilityScope
+                    observability: observabilityScope
```
Comment (tomerd) [inline Sources/CoreCommands/SwiftTool.swift:739]: we generally prefer the verbose form

**Review by tomerd — APPROVED** [review]

File: Sources/CrossCompilationDestinationsTool/ListDestinations.swift [hunk 2]
```diff
@@ -34,7 +34,7 @@ public struct ListDestinations: ParsableCommand {
     public func run() throws {
         let fileSystem = localFileSystem
         let observabilitySystem = ObservabilitySystem.swiftTool()
-        let observabilityScope = observabilitySystem.topScope
+        let scope = observabilitySystem.topScope
```
Comment (tomerd) [inline Sources/CrossCompilationDestinationsTool/ListDestinations.swift:37]: ditto

**Review by tomerd — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Review by Lishkk — APPROVED** [review]

**Pull request merged**